cmd_root.go

```diff
@@ -1,21 +1,55 @@
 package main
 
 import (
+	"fmt"
 	"os"
 
 	"github.com/spf13/cobra"
 
 	libhoney "github.com/honeycombio/libhoney-go"
 )
 
-func commandRoot(cfg *libhoney.Config, filename *string, ciProvider *string) *cobra.Command {
+func commandRoot(cfg *libhoney.Config, filename *string, ciProvider *string, serviceName *string) *cobra.Command {
 	root := &cobra.Command{
 		Version: Version,
 		Use:     "buildevents",
 		Short:   "buildevents creates events for your CI builds",
 		Long: `
 The buildevents executable creates Honeycomb events and tracing information
 about your Continuous Integration builds.`,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			quiet, _ := cmd.Flags().GetBool("quiet")
+			if isClassic(cfg.APIKey) {
+				// if we're in classic mode, we want to behave the same as we always have
+				if cfg.Dataset == "" {
+					cfg.Dataset = "buildevents"
+				}
+				if *serviceName != "" {
+					fmt.Fprintf(os.Stderr, "WARN: classic mode ignores the service name parameter.\n")
+				}
+			} else {
+				// This is the non-classic behavior
+				if *serviceName != "" {
+					// service name was specified, so use it as the dataset
+					if cfg.Dataset != "" && !quiet {
+						// warn if we're going to ignore a specified dataset
+						fmt.Fprintf(os.Stderr, "WARN: service name was specified, dataset is ignored.\n")
+					}
+					cfg.Dataset = *serviceName
```

Comment:
In other libhoney E&S work we have stripped and warn'd if there is whitespace in the dataset name - it would look something like:
```suggestion
					trimmed := serviceName.trim()
					if trimmed != *serviceName {
						fmt.Fprintf(os.stderr, "WARN: service name contained leading or trailing whitespace, sending to '%s'.\n", trimmed)
					}
					cfg.Dataset = trimmed
```